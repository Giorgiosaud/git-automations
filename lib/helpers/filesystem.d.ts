export declare const readFile: (path: string) => Promise<any>;
export declare const readEnv: (path: string) => Promise<any>;
