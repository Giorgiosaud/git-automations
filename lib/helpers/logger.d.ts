export declare const PRE_PROCESSED: (...message: string[]) => void;
export declare const INFO: (...message: string[]) => void;
export declare const PROCESSED: (...message: string[]) => string;
export declare const WARNING: (...message: string[]) => string;
export declare const NORMAL: (...message: string[]) => string;
export declare const ERROR: (...message: string[]) => void;
