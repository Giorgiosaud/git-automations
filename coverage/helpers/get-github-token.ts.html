
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for helpers/get-github-token.ts</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../index.html">All files</a> / <a href="index.html">helpers</a> get-github-token.ts</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">85% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>17/20</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>2/2</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>1/1</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">85% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>17/20</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a>
<a name='L28'></a><a href='#L28'>28</a>
<a name='L29'></a><a href='#L29'>29</a>
<a name='L30'></a><a href='#L30'>30</a></td><td class="line-coverage quiet"><span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import * as  path from 'node:path'
import {homedir} from 'node:os'
import {readEnv, buildEnvContent} from './file-system'
import axios from 'axios'
import {CliUx} from '@oclif/core'
import {writeFileSync, existsSync} from 'node:fs'
export default async (rcPath: string, organization?: string): Promise&lt;string&gt; =&gt; {
  const testTokenUrl = `https://api.github.com/orgs/${organization}/repos?type=member`
  const rcRealPath = path.resolve(homedir(), rcPath)
  const SETTINGS_FILE_EXIST = await existsSync(rcRealPath)
  if (!SETTINGS_FILE_EXIST) {
    writeFileSync(rcRealPath, '')
  }
&nbsp;
  const SETTINGS = await readEnv(rcRealPath)
  if (!SETTINGS.GITHUB_TOKEN) {
    const token = await CliUx.ux.prompt('Please add your repository token')
    await axios.get(testTokenUrl, {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    })
    const newEnv = <span class="cstat-no" title="statement not covered" >buildEnvContent({GITHUB_TOKEN: token})</span>
<span class="cstat-no" title="statement not covered" >    writeFileSync(path.resolve(rcRealPath), newEnv)</span>
<span class="cstat-no" title="statement not covered" >    return token</span>
  }
&nbsp;
  return SETTINGS.GITHUB_TOKEN
}
&nbsp;</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Tue Nov 29 2022 20:59:03 GMT-0300 (Chile Summer Time)
            </div>
        <script src="../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../sorter.js"></script>
        <script src="../block-navigation.js"></script>
    </body>
</html>
    