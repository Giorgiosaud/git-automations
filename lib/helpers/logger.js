"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var chalk_1 = __importDefault(require("chalk"));
var log = console.log;
exports.PRE_PROCESSED = function () {
    var message = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        message[_i] = arguments[_i];
    }
    return log(chalk_1.default.bgBlue.whiteBright(message));
};
exports.INFO = function () {
    var message = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        message[_i] = arguments[_i];
    }
    return log(chalk_1.default.bgBlue.whiteBright(message));
};
exports.PROCESSED = function () {
    var message = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        message[_i] = arguments[_i];
    }
    return chalk_1.default.bgGreen.whiteBright(message);
};
exports.WARNING = function () {
    var message = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        message[_i] = arguments[_i];
    }
    return chalk_1.default.bgYellow.redBright(message);
};
exports.NORMAL = function () {
    var message = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        message[_i] = arguments[_i];
    }
    return chalk_1.default.bgBlack.whiteBright(message);
};
exports.ERROR = function () {
    var message = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        message[_i] = arguments[_i];
    }
    return log(chalk_1.default.bgRed.yellowBright(message));
};
