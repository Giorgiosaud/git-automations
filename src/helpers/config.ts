export const rcPath = '.github-automation.yml'
export default {
  rcPath,
}
