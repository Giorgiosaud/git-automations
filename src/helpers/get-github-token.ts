import {homedir} from 'node:os'
import {readEnv, buildEnvContent} from './file-system'
import axios from 'axios'
import {ux} from '@oclif/core'
import {writeFileSync, existsSync} from 'node:fs'
import {resolve} from 'node:path'

export default async (rcPath: string, organization?: string): Promise<string> => {
  const testTokenUrl = `https://api.github.com/orgs/${organization}/repos?type=member`
  const rcRealPath = resolve(homedir(), rcPath)
  console.log(rcRealPath)

  const SETTINGS_FILE_EXIST = await existsSync(rcRealPath)
  if (!SETTINGS_FILE_EXIST) {
    writeFileSync(rcRealPath, '')
  }

  const SETTINGS = await readEnv(rcRealPath)
  if (!SETTINGS.GITHUB_TOKEN) {
    const token = await ux.prompt('Please add your repository token')
    await axios.get(testTokenUrl, {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    })
    const newEnv = buildEnvContent({GITHUB_TOKEN: token})
    writeFileSync(resolve(rcRealPath), newEnv)
    return token
  }

  return SETTINGS.GITHUB_TOKEN
}
