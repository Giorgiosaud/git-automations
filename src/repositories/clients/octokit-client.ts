import { Octokit } from "octokit"

export default function ({ auth }: { auth: string} ): Octokit {
  return new Octokit({
    auth,
  })
}