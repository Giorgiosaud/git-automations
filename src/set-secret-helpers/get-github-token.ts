import {readEnv, buildEnvContent, writeFile} from '../helpers/filesystem'
import * as  path from 'path'
import inquirer from 'inquirer'
import cli from 'cli-ux'
import {homedir} from 'os'

export default async (): Promise<string> => {
  const SETTINGS = await readEnv(path.resolve(homedir(), '.githubtoken'))
  if (!SETTINGS.GITHUB_TOKEN) {
    const Answers = await cli.prompt('please add your github token to the base settings')
    const newEnv = buildEnvContent({GITHUB_TOKEN: Answers.token})
    await writeFile(path.resolve(homedir(), '.githubtoken'), newEnv)
    return Answers.token
  }
  return SETTINGS.GITHUB_TOKEN
}
