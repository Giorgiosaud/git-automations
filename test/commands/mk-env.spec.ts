import {expect, test} from '@oclif/test'
import { assert, stub } from 'sinon';

import * as getGithubToken from '../../src/helpers/get-github-token'
import * as octokitClient from '../../src/repositories/clients/octokit-client'

const envRes = {
  data: {
    environments: [{
      id: 1,
      name: 'pepe',
    }]
  }
}

const octokitClientStub={
  request:stub()
}

describe('Testing mk-env command', () => {
  beforeEach(() => {
    octokitClientStub.request.reset()
  })
  test
  .command(['mk-env'])
  // .exit(0)
  .catch(error => {
    expect(error.message).to.contain('The following errors occurred:')
    expect(error.message).to.contain('environments')
    expect(error.message).to.contain('organization')
    expect(error.message).to.contain('repositories')
  })
  .it('mk-env fails if no flags are set asking for repo')
  
  test.stdout()
  .command(['mk-env','-e', 'env'])
  // .exit(0)
  .catch(error => {
    expect(error.message).to.contain('The following errors occurred:')
    expect(error.message).to.contain('organization')
    expect(error.message).to.contain('repositories')
  })
  .it('mk-env fails if only env flag is set')
  
  test.stdout()
  .command(['mk-env','-e', 'env','-o', 'organization'])
  // .exit(0)
  .catch(error => {
    expect(error.message).to.contain('The following error occurred:')
    expect(error.message).to.contain('repositories')
  })
  .it('mk-env fails if only env and org flags are set')
  
  
  test.stdout()
  .stub(getGithubToken,'default',stub=>stub.returns('token'))
  .command(['mk-env','-e', 'env','-o', 'organization','-r', 'rep sso'])
  .catch(error => {
    expect(error.message).to.contain('The repository string must only contain numbers leters and dash')
  })
  .it('mk-env fails if repo is not well formatted')
  
  test
  .stdout()
  .do(()=>{
    octokitClientStub.request.onFirstCall().returns(Promise.resolve(envRes))
  })
  .stub(octokitClient,'default',stub=>stub.returns(octokitClientStub))
  .stub(getGithubToken,'default',stub=>stub.returns('token'))
  .command(['mk-env','-e', 'env','-o', 'organization','-r', 'repo-1'])
  .it('mk-env works if repo name match',()=>{
    expect(octokitClientStub.request.calledTwice).to.be.true
    assert.calledWithExactly(octokitClientStub.request.firstCall,'GET /repos/{owner}/{repo}/environments', {headers: {'X-GitHub-Api-Version': '2022-11-28'}, owner: 'organization' , repo: 'repo-1'})
    assert.calledWithExactly(octokitClientStub.request.secondCall,'PUT /repos/{owner}/{repo}/environments/{environment_name}', {environment_name: 'env', headers: {'X-GitHub-Api-Version': '2022-11-28'},owner: 'organization' , repo: 'repo-1'})
  })

  test
  .stdout()
  .do(()=>{
    octokitClientStub.request.returns(Promise.resolve(envRes))
  })
  .stub(octokitClient,'default',stub=>stub.returns(octokitClientStub))
  .stub(getGithubToken,'default',stub=>stub.returns('token'))
  .command(['mk-env','-e', 'env','-o', 'organization','-r', 'repo-1','repo-2'])
  .it('mk-env works if multi repositories set',()=>{
    expect(octokitClientStub.request.callCount).to.be.eq(4)
    assert.calledWithExactly(octokitClientStub.request.getCall(0), 'GET /repos/{owner}/{repo}/environments', {headers: {'X-GitHub-Api-Version': '2022-11-28'}, owner: 'organization', repo: 'repo-1'})
    assert.calledWithExactly(octokitClientStub.request.getCall(1), 'PUT /repos/{owner}/{repo}/environments/{environment_name}', {environment_name: 'env', headers: {'X-GitHub-Api-Version': '2022-11-28'}, owner: 'organization', repo: 'repo-1'})
    assert.calledWithExactly(octokitClientStub.request.getCall(2), 'GET /repos/{owner}/{repo}/environments', {headers: {'X-GitHub-Api-Version': '2022-11-28'}, owner: 'organization', repo: 'repo-2'})
    assert.calledWithExactly(octokitClientStub.request.getCall(3), 'PUT /repos/{owner}/{repo}/environments/{environment_name}', {environment_name: 'env', headers: {'X-GitHub-Api-Version': '2022-11-28'}, owner: 'organization', repo: 'repo-2'})
  })
  test
  .stdout()
  .do(()=>{
    octokitClientStub.request.returns(Promise.resolve(envRes))
  })
  .stub(octokitClient,'default',stub=>stub.returns(octokitClientStub))
  .stub(getGithubToken,'default',stub=>stub.returns('token'))
  .command(['mk-env','-e', 'env-1','env-2','-o', 'organization','-r','repo-2'])
  .it('mk-env works if multi environments set',()=>{
    expect(octokitClientStub.request.callCount).to.be.eq(3)
    // assert.calledWithExactly(octokitClientStub.request.getCall(0), 'GET /repos/{owner}/{repo}/environments', {headers: {'X-GitHub-Api-Version': '2022-11-28'}, owner: 'owner', repo: 'repo-1'})
    // assert.calledWithExactly(octokitClientStub.request.getCall(1), 'PUT /repos/{owner}/{repo}/environments/{environment_name}', {environment_name: 'env-1', headers: {'X-GitHub-Api-Version': '2022-11-28'}, owner: 'owner', repo: 'repo-1'})
    // assert.calledWithExactly(octokitClientStub.request.getCall(2), 'PUT /repos/{owner}/{repo}/environments/{environment_name}', {environment_name: 'env-2', headers: {'X-GitHub-Api-Version': '2022-11-28'}, owner: 'owner', repo: 'repo-1'})
  })

  test
  .stdout()
  .do(()=>{
    octokitClientStub.request.returns(Promise.resolve({data: {
      environments: [{
        id: 1,
        name: 'env-1',
      }]}
    }))
  })
  .stub(octokitClient,'default',stub=>stub.returns(octokitClientStub))
  .stub(getGithubToken,'default',stub=>stub.returns('token'))
  .command(['mk-env','-e', 'env-1', 'env-2', '-o', 'organization', '-r', 'repo-1'])
  .it('mk-env works if multi environments set but not call repeated one',()=>{
    expect(octokitClientStub.request.callCount).to.be.eq(2)
    assert.calledWithExactly(octokitClientStub.request.getCall(0), 'GET /repos/{owner}/{repo}/environments', {headers: {'X-GitHub-Api-Version': '2022-11-28'}, owner: 'organization', repo: 'repo-1'})
    assert.calledWithExactly(octokitClientStub.request.getCall(1), 'PUT /repos/{owner}/{repo}/environments/{environment_name}', {environment_name: 'env-2', headers: {'X-GitHub-Api-Version': '2022-11-28'}, owner: 'organization', repo: 'repo-1'})
  })

  
})
