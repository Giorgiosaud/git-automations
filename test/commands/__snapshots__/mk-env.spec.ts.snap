// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mk-env command mk-env fails if no flags are set asking for repo 1`] = `
"The following errors occurred:
  [2mMissing required flag environments[22m
[2m  Missing required flag organization[22m
[2m  Missing required flag repositories[22m
See more help with --help"
`;

exports[`mk-env command mk-env fails if only env and org is set 1`] = `
"The following error occurred:
  [2mMissing required flag repositories[22m
See more help with --help"
`;

exports[`mk-env command mk-env fails if only env is set 1`] = `
"The following errors occurred:
  [2mMissing required flag organization[22m
[2m  Missing required flag repositories[22m
See more help with --help"
`;

exports[`mk-env command mk-env fails if repo name not match 1`] = `"The repository string must only contain numbers leters and dash"`;
